crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -401,6 +408,42 @@ impl ImageFolderDataset {
         Self::with_items(items, classes)
     }
 
+    /// Create an image segmentation dataset with the specified items.
+    ///
+    /// # Arguments
+    ///
+    /// * `items` - List of dataset items, each item represented by a tuple `(image path, labels)`.
+    /// * `classes` - Dataset class names.
+    ///
+    /// # Returns
+    /// A new dataset instance.
+    pub fn new_segmentation_with_items<P: AsRef<Path>, S: AsRef<str>>(
+        items: Vec<(P, SegmentationMask)>,
+        classes: &[S],
+    ) -> Result<Self, ImageLoaderError> {
```

Comment:
With the suggested changes from my previous comment, it would make more sense (and probably be more practical from a user standpoint) to have a `new_segmentation` method that instead takes a list of `(image path, mask path)` pairs (so `Vec<(P, P)>`).

We still need the method to accept the class names, which can be used to map to a class id similar to what is already done. These identifiers would be use to map a pixel value to a class.

For added flexibility we could provide a way to have specific pixel values map to a class.